bootloader/src/pyi_win32_utils.c

```diff
@@ -435,6 +435,55 @@ pyi_win32_utf8_to_mbs(char * dst, const char * src, size_t max)
     }
 }
 
+
+/* Retrieve the SID of the current user.
+ *  Returns SID string on success, NULL on failure.
+ */
+static char *
+pyi_win32_get_user_sid()
+{
+    HANDLE process_token;
+    DWORD user_info_size, result;
+    PTOKEN_USER user_info;
+    wchar_t *wsid = NULL;
+    char *sid;
+
+    // Get access token for the calling process
+    if (!OpenProcessToken(GetCurrentProcess(), TOKEN_QUERY, &process_token)) {
+        return NULL;
+    }
+
+    // Get buffer size and allocate buffer
+    if (!GetTokenInformation(process_token, TokenUser, NULL, 0, &user_info_size)) {
+        if (GetLastError() != ERROR_INSUFFICIENT_BUFFER) {
+            CloseHandle(process_token);
+            return NULL;
+        }
+    }
+
+    user_info = (PTOKEN_USER)calloc(1, user_info_size);
+
+    // Get user information
+    if (!GetTokenInformation(process_token, TokenUser, user_info, user_info_size, &user_info_size)) {
+        free(user_info);
+        CloseHandle(process_token);
+        return NULL;
+    }
+
+    // Convert SID to string
+    ConvertSidToStringSidW(user_info->User.Sid, &wsid);
```

Comment:
Check missing whether this succeeded.

Reply:
In the revised version, this call is the last before cleanup and return, so check is not needed anymore.